src/python/gem5/resources/resource.py

```diff
@@ -122,6 +128,8 @@ class FileResource(AbstractResource):
     def __init__(
         self,
         local_path: str,
+        id: Optional[str] = None,
+        category: Optional[str] = None,
```

Comment:
Instead of having this as a parameter why not have the `AbstractResource` class enforce the creation of a `def get_category_name(cls) -> str` function. Then each of the categories has this implemented.